### PR TITLE
convert JSON empty strings to zero values for all SPL numeric types

### PR DESCRIPTION
**Pull request by ejpring**

### CODE REVIEW

**Comment by ddebrunner** [comment]
So it seems Number("") in JavaScript does evaluate to zero.

If this approach is seen as acceptable then I think it should be handled the same for all numeric types.

Also String has a nifty `isEmpty()` method.

**Comment by ddebrunner** [comment]
You can try with this:
http://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_number

**Comment by ddebrunner** [comment]
Also here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number

**Comment by ejpring** [comment]
I've finished fixing jsonToAttribute() to convert empty JSON strings to zero values for all SPL numeric types, and added a testcase to verify them. I think this code is ready for the master branch.

File: com.ibm.streamsx.json/impl/java/src/com/ibm/streamsx/json/converters/JSONToTupleConverter.java [hunk 1]
```diff
@@ -41,48 +41,48 @@
...
+		if (l.isLoggable(TraceLevel.DEBUG)) {
+          l.log(TraceLevel.DEBUG, "Converting Java value '" + jsonObj.toString() + "' of type " + jsonObj.getClass().getSimpleName() + " to SPL attribute " + name + " of type " + type.toString());
 		}
-		if(jsonObj == null) return null;
 		try {
 			switch(type.getMetaType()) {
+			case BOOLEAN:
+				if(jsonObj instanceof Boolean)
+					return (Boolean)jsonObj;
+				return jsonObj.toString().isEmpty() ? false : Boolean.parseBoolean(jsonObj.toString());
```
Comment (ddebrunner) [inline com.ibm.streamsx.json/impl/java/src/com/ibm/streamsx/json/converters/JSONToTupleConverter.java]: Minor nit that simply `Boolean.parseBoolean(jsonObj.toString());` is sufficient

**Comment by ejpring** [comment]
Yep -- that simplification produces the same results. Thanks!

**Pull request merged**